### DISCUSSION TITLE
gulp file order

### DISCUSSION THREAD

**Discussion by t1a2l** [gulpjs/gulp#2451] [Help]
gulp file order is different between windows and linux. on linux it is working as intended but windows mix up the file order and cause the app not to work. (there are about 100 files - ordering manually is not an option).

**Comment by phated** [comment]
Sorting a stream is extremely expensive/slow, so we attempt to push files in the order they are globbed from your filesystem. Since filesystems behave differently, that is likely the cause of this issue for you.

Since you require them to be in a specific order, the performance penalty probably is less of an issue for you. I recommend using https://www.npmjs.com/package/gulp-sort to sort your files before you pass them to the plugin where they need to be ordered in a specific way.

**Reply by t1a2l** [reply]
It seems that the issue after some investigating is the underscore. In windows it is normal: (asc)
app\models\shipments_filter.js
app\models\shipment_history.js
app\models\shipment_multi.js

while on linux the underscore is making gulp ignore "asc"
app/models/shipment_history.js
app/models/shipment_multi.js
app/models/shipments_filter.js

Is it a bug?
@phated 

Thanks anyway.
